x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/query/ScoreScriptUtils.java

```diff
@@ -74,20 +105,21 @@ public void validateDocVector(BytesRef vector) {
                 throw new IllegalArgumentException("The query vector has a different number of dimensions [" +
                     queryVector.length + "] than the document vectors [" + vectorLength + "].");
             }
+            return vector;
```

Comment:
Not related to this PR, but I think we can do the vectorLength  validation on lines 103-108 just for the first doc, and then set a flag `validated` = true, and skip this validations for further docs.

Reply:
@rjernst yes, we guarantee at indexing time that all vector fields have the same length. Specifically, there is a mapping parameter `dims`, and all vector fields must match that number of dimensions.

Earlier I ran benchmarks to measure the cost of length validation, and it was very small (couldn't detect it in the results). So I'm not sure it's worth adding an extra guard + check here for performance. But if the scripting framework had a way to perform a one-time validation, it could make this code a bit cleaner.